### ISSUE TITLE
yyaxis confuses LaTeX

### ISSUE THREAD

**Issue by Pratched** [matlab2tikz/matlab2tikz#905]
hi, im trying to implement some figures from Matlab via matlab2tikz into LaTeX. But while runnig
```
\begin{figure}
  \input{....tikz}
\end{figure}
```
I do get a few errors containing: 'dimension too large', 'cant work with sizes bigger than 19feet', etc.
I do realize that this is probably a problem made up by LaTeX but when creating the pdf file, the actual figure appears. Well I could ignore the errors for now, but the figure should have a left and a right y axis. Which I createt via `yyaxis` in Matlab. 
It appears to me, that the decision which y axis will be plotted in LaTeX depends on whether I executed `yyaxis left` or `yyaxis right` at last. Beacause I always do get the same error code, but on different lines, depending the last yyaxis in Matlab. And so is the plotted y axis changing.
Is there any way (Matlab, but maybe you even do have the solution for LaTeX) I can (easily) get both y axis plotted?

I do have some basic skills with Matlab and LaTeX but that's it. So I i apologize in advanced for some foolish mistakes.

**Comment by miscco** [comment]
Hi can you please give us a minimal working example?

I have never used the yyaxis function myself, but only plotyy. Maybe that would be a workaround for you?

**Comment by miscco** [comment]
As a side not i would like to mention, that this function has been introduced in Matlab2016a

I dont know if we already support those new function. In any case i only have 2015a so i cannot test this.

**Comment by Pratched** [comment]
Thanks for the fast repley.
Well I actually started with plotyy, but it was way more easier to 
```
hold on; 
plot( x, y1, x, y2);
yyaxis rigth;
plot(x, y3);
```
than try do plot 3 graphs with `plotyy`.

Well, I am using the student version(2016a since a few weeks) and Mathworks.com (or so) suggested to use yyaxis instead of plotyy. No I can see, why I cant find very much information about yyaxis.
I am not quit sure what do you mena by working example. Should I post the Matlab code itself?

**Comment by Pratched** [comment]
Maybe it is importent that x is date. I did implement `datenum` above `hold on`.
```
x=datenum(x);
hold on; 
plot( x, y1, x, y2);
yyaxis rigth;
plot(x, y3);
datetick;
```
x was actualy a date with the format: dd.mm.yyyy HH:SS. Since the Date for all 177 lines of x is the same, after `datetick` only the tim (HH:SS) appears at the x axis.
Maybe this is important as well.

**Comment by miscco** [comment]
You can easily plot multiple plots with plotyy
```matlab
x=linspace(0, 10)
y1 = sin(3*x);
y2=sin(4*x);
z = sin(3*x).*exp(0.5*x);

plotyy(x, [y1; y2], x, z);
```

You can also easily expand that by defining your own custom function. What i did to show a comparison of mean and std from experimental data vs model output is the following (using boundedline from mathworks):

```matlab
BL_model =@(y,x) boundedline(y,x(:,1), x(:,2), 'alpha', 'transparency', 0.1, 'r');
BL_data  =@(y,x) boundedline(y,x(:,1), x(:,2), 'alpha', 'transparency', 0.1, 'black');

plotyy(time, [mean_data, std_data], time, [mean_model, std_model], BL_data, BL_model);
```

**Comment by okomarov** [comment]
We currently do not support the `yyaxis()` function yet. Please, use the workaround suggested by @miscco.

For developers, the axis produced with `yyaxis()` has two rulers, left and right:
```
h = 
  Axes with properties:

    YAxisLocation: 'right'
            YAxis: [2x1 NumericRuler]
             YLim: [-150 100]
             XLim: [0 10]
           XScale: 'linear'
           YScale: 'linear'
         Position: [0.1300 0.1100 0.7750 0.8150]
            Units: 'normalized'

>> h.YAxis(1)
ans = 
  NumericRuler with properties:

             Limits: [-1 1]
              Scale: 'linear'
           Exponent: 0
         TickValues: [-1 -0.8000 -0.6000 -0.4000 -0.2000 0 0.2000 0.4000 0.6000 0.8000 1]
    TickLabelFormat: '%g'

>> h.YAxis(2)
ans = 
  NumericRuler with properties:

             Limits: [-150 100]
              Scale: 'linear'
           Exponent: 0
         TickValues: [-150 -100 -50 0 50 100]
    TickLabelFormat: '%g'
```

**Comment by Pratched** [comment]
Alright, thank you very much, thats the key. Didnt know that.

But, when I do this, there are a few Problems coming up I could avoid using `yyaxis`. Thats probably why mathworks.com is recommending `yyaxis` instead of `plotyy`.

A)
When I plot 3 graphs (2 with the axis on the left side and 1 on the right side) like in your example, matlab keeps moving the graph (x,z)
I dont know why, but if I do as you(miscco) suggested, Matlab keeps moving the xz-graph a bit to the right side. Meaning the 2 graphs of the left side and the right side using probably different xaxis.
I atached 2 pictures where will see the different I am talking about. .._Figure.png is the right figure, plotted with `yyaxis` and ..._Figure#2.png is plotted with `plotyy`. The yellow graph is using the right yaxis.

B)
There was an easy way to change the color of the right yaxis with `yyaxis`.
```
ax=gca;
ax.YColor=[0,0,0];
```
I dont realy know how I can get this done with `plotyy` since I dont refer to the right yaxis.

C)
Probably the same solution as B). How do I label the right yaxis?

Thanks in advanced. the first problem is acutaly the most important. I realize that i probably will find the answers to B) and C) somewhere.

**Comment by miscco** [comment]
Yeah, I think the reason why matlab is suggesting this new functionality is exactly that it is easier to access the properties. However, it can also be accessed with plotyy

Consider the previous example
```matlab
x=linspace(0, 10)
y1 = sin(3*x);
y2=sin(4*x);
z = sin(3*x).*exp(0.5*x);

[axes, Plot1, Plot2] =  plotyy(x, [y1; y2], x, z);
```

Now you can access the axes properties of the individual axes via

```matlab
set(ax(1), 'YLim', [-2, 2])
set(ax(2), 'YLim', [-4, 4])
```

**Comment by Pratched** [comment]
Mh, I am sorry, but I dont understand what exactly you ar doing. I need do move the yellow graph in the second figure back. Because, as you can see in the first figure, it has to end at "18:00", which is does not in the second figure. Im confused becaus I actualy are using the same Date-Vector for the xaxis. "x" is a vectore which is mostly the same number  only different in the 4th do 6th decimal place. The figure shows the time, because I used `datetick`.
Why is Matlab moving the yellow graph and what is meant by 'Ylim'?

**Comment by miscco** [comment]
So YLim gives the limits of the corresponding Y-Axis. So if you want to change those limits to [y_low, y_high] and also set the limits of the X-Axis you have to do:
```
[ax, Plot1, Plot2] =  plotyy(x, [y1; y2], x, z);

set(ax(2), 'XLim',  get(ax(1), 'XLim'), 'YLim',  [y_low, y_high]);
```

**Comment by miscco** [comment]
@Pratched BTW the reason why MATLAB is moving the second plot seems to be that there are less data points? I am not 100% positive, but it seems that your first two plots have some area where they are 0?

**Comment by miscco** [comment]
@okomarov Do you have any suggestions what we could do to support it? It doesnt seem to be that much different from plotyy? 

Maybe we have to adopt isAxisPlotyy?

**Comment by Pratched** [comment]
Okay. I will try to get something done, but I do not understand WHY I have to do something at all?
It is the same variable for the xaxis. and If I plot only z with 
```
plot(x,z);
```
I do get the right graph.

**Comment by okomarov** [comment]
@miscco It is not complicated, it is just a lot of work. Apart from testing wether it has a second axis, we need to retrieve properties that we didn't in the past. 

I reckon as a general design approach, if we want to dig deeper into HG2, it would be easier to start working on m2t_hg2, rather than patching the code.

**Comment by Pratched** [comment]
Yes u are right. But it worked with `yyaxis` so I thought this will work with `plotyy`as well. 
x, y1, y2 and z are a one-row-matrix with 177 lines. z= y1.*y2 so z does have some lines filled with "NaN" because in these lines y1 and y2 have  zeros.

**Comment by miscco** [comment]
@Pratched I think Matlab is actually reasonable here, but in an unfriendly way.

If i understand it correctly, plotyy creates two independent plots, which includes the x axis, whereas 
```
hold on
plot
pot
hold off
```
Plots everything in the same axis. This has the side effect, that the x-range is taken as the largest one. However, in the other case each plot has its individual x-range. I would suggest, that you simply do
```matlab
z(isnan(z))=0;
```
To set all NaNs to zero, which should be legitimate if i understand your plot correctly.

**Comment by Pratched** [comment]
Okay, thanks. That seems to be a way. I will try and let you know.

**Comment by miscco** [comment]
Yeah, dividing by zero still means a "Wirkungsgrad" of zero

**Comment by Pratched** [comment]
Well as you can see I`ve got the NaN's to zero but the problem is not solved.

**Comment by Pratched** [comment]
But you could also argue, that the there is no "Wirkungsgrad" where there is no "Leistung". ;)

**Comment by miscco** [comment]
@Pratched Where can i see that it didnt work? Please include the latest figure and keep the old ones for history. Otherwise it is hard to follow the thread in a week or so.

**Comment by Pratched** [comment]
![aufg3_figure 2](https://cloud.githubusercontent.com/assets/19487828/15430418/9740f6ee-1ea5-11e6-8908-5db3f54fa0fa.png)
![aufg3_figure 4](https://cloud.githubusercontent.com/assets/19487828/15430417/973818da-1ea5-11e6-87e1-2d222a674b23.png)

first one without "NaN's to zero, second with.

**Comment by miscco** [comment]
Can you please also post the code you are using. It is hard to just tell from the figure.

**Comment by Pratched** [comment]
Alright.

```
P_DCinmW=I_PVinmA.*(U_PVinV);
P_ACinmW=I_NetzinmA.*(U_NetzinV);
P_DCinW=P_DCinmW./1000;
P_ACinW=P_ACinmW./1000;

TimeA=datenum(Time); 
n_WR_real(isnan(n_WR_real))=0;

plotyy(TimeA, [P_DCinW, P_ACinW], TimeA, n_WR_real);

ylabel('Leistung [W]');
ylabel('\eta_{WR,real}');

datetick;
grid on;

legend('P_{DC}','P_{AC}','\eta_{WR,real}'); xlabel('Tagesverlauf am 27.04.2016'); 
title('Aufgabe 3.1
```

**Comment by miscco** [comment]
Can you try the following
```matlab

[ax, ~ ~]= plotyy(TimeA, [P_DCinW, P_ACinW], TimeA, n_WR_real);
set(ax(2), 'XLim', get(ax(1), 'XLim'))
```

**Comment by Pratched** [comment]
Thanks. I did 
´´´
TimeA=datenum(Time); 
%n_WR_real(isnan(n_WR_real))=0;

[ax, ~ ~]= plotyy(TimeA, [P_DCinW, P_ACinW], TimeA, n_WR_real);
set(ax(2), 'XLim', get(ax(1), 'XLim'))
%plotyy(TimeA, [P_DCinW, P_ACinW], TimeA, n_WR_real);

ylabel('Leistung [W]');
ylabel('\eta_{WR,real}');

datetick;
grid on;

legend('P_{DC}','P_{AC}','\eta_{WR,real}'); xlabel('Tagesverlauf am 27.04.2016'); 
title('Aufgabe 3.1');
```
But there was an error: "Error: File: Aufg3_script.m Line: 12 Column: 9
Unbalanced or unexpected parenthesis or bracket."
Line 12 is where I implemented your code.(There is some other code above just always calculating the variables)

**Comment by okomarov** [comment]
@Pratched To solve more effectively your issue, please provide the data, i.e. `Time, P_DCinW, P_ACinW, n_WR_real`.

You can save the variables in a .mat  file and drop them here.

**Comment by miscco** [comment]
You are missing a comma in the [ax, ~, ~]

**Comment by Pratched** [comment]
Alright. Just give me a minute I found something out.

**Comment by Pratched** [comment]
I added the comma but it didnt changed anything. but thanks for trying. I did found out what the problem is here.
(http://de.mathworks.com/matlabcentral/newsreader/view_thread/304888)
`datetick`does something funny. I skipped `datetick` and got the right graph, but I am missing the time :(
![aufg3_figure 5](https://cloud.githubusercontent.com/assets/19487828/15431909/3959c338-1eac-11e6-9f86-2d5731ced798.png)

**Comment by miscco** [comment]
@Pratched I was wondering about that too, because i got the right plot and then added datetick.

**Comment by Pratched** [comment]
But how do I get the time instead of the serial date numbers without datetick

**Comment by miscco** [comment]
@Pratched Ah actually the solution is simple

```matlab
datetick(ax(1))
datetick(ax(2))
```

**Comment by Pratched** [comment]
Mh just fill ax() with 1 and 2?
I do get the error

> Error using datetick>parseinputs (line 360)
The axis must be 'x','y', or 'z'.

Error in datetick (line 108)
[axh,nin,ax,dateform,keep_ticks,keep_limits] = parseinputs(varargin);

Error in Aufg3_script (line 19)
datetick(ax(1)); datetick(ax(2));
 <

**Comment by miscco** [comment]
No you have use the plot command i gave you
```matlab
[ax, ~ ,~]= plotyy(TimeA, [P_DCinW, P_ACinW], TimeA, n_WR_real);
datetick(ax(1))
datetick(ax(2))
```

**Comment by Pratched** [comment]
Thank you very much! Now I just need to understand what exactly your code does an adapt the axes and formation. (Would like to have some more values on the left yaxis).
But I am sure the web will provide this informations ;)

The only problem is: LaTeX does not acceot this either ...

**Comment by miscco** [comment]
@Pratched You have to check the manual for plotyy. It will tell you, that [ax, ~ , ~] = plotyy returns the two axes handles for the respective plots. 

regarding the tex code, did you use the current develop branch from m2t?

**Comment by Pratched** [comment]
I did load this files: 
"http://www.mathworks.com/matlabcentral/fileexchange/22022-matlab2tikz-matlab2tikz"
I can see the figure in the pdf-file but the error do come up again. I guess I have to live with these errors. I did got the actual result.

**Comment by miscco** [comment]
You might want to download from here
https://github.com/matlab2tikz/matlab2tikz/tree/develop

**Comment by miscco** [comment]
Hm there seems to be a problem with the dateticks.

**Comment by okomarov** [comment]
@miscco @Pratched We have 2 problems here:

1.  We currently do not support new HG2 features like `yyaxis()`, for which we have the `plotyy()`workaround
2. We do not support dates in the Matlab format, i.e. 736447.166666667, and this is how they appear in the .tex, thus creating `Dimension too large. \end{axis} errors`

Now, we should open two separate issues. The second issue can be tackled by re-indexing the dates, i.e.

**Comment by Pratched** [comment]
Mh, didnt change very much.
In case someone is interessted in the error from LaTeX. I have got the pdf so I am satisfied for no. Thank you again, I will have a look at the manual for plotyy.

> >! Dimension too large.
> <recently read> \pgf@xx
> l.398 \end{axis}
> I can't work with sizes bigger than about 19 feet.
> Continue and I'll use the largest value I can.
> ! Undefined control sequence.
> \pgfplots@label@ ->$\eta {}_{\text
> {WR,real}}$
> l.398 \end{axis}
> The control sequence at the end of the top line
> of your error message was never \def'ed. If you have
> misspelled it (e.g., `\hobx'), type `I' and the correct
> spelling (e.g., `I\hbox'). Otherwise just continue,
> and I'll forget about whatever was undefined.
> ! Undefined control sequence.
> <recently read> \text
> l.398 \end{axis}
> The control sequence at the end of the top line
> of your error message was never \def'ed. If you have
> misspelled it (e.g., `\hobx'), type `I' and the correct
> spelling (e.g., `I\hbox'). Otherwise just continue,
> and I'll forget about whatever was undefined.
> ! Undefined control sequence.
> \pgfplots@already@computed@legend@node ..._{\text
> {DC}}$};\\\pgfplots@show@s...
> l.398 \end{axis}
> The control sequence at the end of the top line
> of your error message was never \def'ed. If you have
> misspelled it (e.g., `\hobx'), type `I' and the correct
> spelling (e.g., `I\hbox'). Otherwise just continue,
> and I'll forget about whatever was undefined.
> ! Undefined control sequence.
> <recently read> \text
> l.398 \end{axis}
> The control sequence at the end of the top line
> of your error message was never \def'ed. If you have
> misspelled it (e.g., `\hobx'), type `I' and the correct
> spelling (e.g., `I\hbox'). Otherwise just continue,
> and I'll forget about whatever was undefined.
> ! Undefined control sequence.
> \pgfplots@already@computed@legend@node ..._{\text
> {AC}}$};\\};
> l.398 \end{axis}
> The control sequence at the end of the top line
> of your error message was never \def'ed. If you have
> misspelled it (e.g., `\hobx'), type `I' and the correct
> spelling (e.g., `I\hbox'). Otherwise just continue,
> and I'll forget about whatever was undefined.
> Missing character: There is no ; in font nullfont!
> ! Dimension too large.
> <recently read> \pgf@xx
> l.599 \end{axis}
> I can't work with sizes bigger than about 19 feet.
> Continue and I'll use the largest value I can.
> ! Undefined control sequence.
> \pgfplots@already@computed@legend@node ..._{\text
> {WR,real}}$};\\};
> l.599 \end{axis}
> The control sequence at the end of the top line
> of your error message was never \def'ed. If you have
> misspelled it (e.g., `\hobx'), type `I' and the correct
> spelling (e.g., `I\hbox'). Otherwise just continue,
> and I'll forget about whatever was undefined.
> )
> ! I can't write on file `PraktikumPhotovoltaik.pdf'.
> Please type another file name for output
> ! Emergency stop.
> \@EveryShipout@Output ...@Org@Shipout \box \@cclv
> l.63 \end{figure}
> *** (job aborted, file error in nonstop mode)
> Here is how much of TeX's memory you used:
> 23330 strings out of 493634
> 600988 string characters out of 3143738
> 916722 words of memory out of 3000000
> 26368 multiletter control sequences out of 15000+200000
> 6342 words of font info for 23 fonts, out of 3000000 for 9000
> 1025 hyphenation exceptions out of 8191
> 73i,19n,84p,8529b,2158s stack positions out of 5000i,500n,10000p,200000b,50000s
> ! ==> Fatal error occurred, no output PDF file produced!

**Comment by miscco** [comment]
@okomarov Is there any way to determine dateticks are used?

**Comment by okomarov** [comment]
@miscco for `datetime()` yes, otherwise we need to create a FAQ with a workaround that shows how to re-index dates, basically with a `unique()` and its 3rd output argument

**Comment by Pratched** [comment]
sorry, I did something ^without intention to do anything at all. but not quite sure what

**Comment by okomarov** [comment]
@Pratched See if this newly created [FAQ](https://github.com/matlab2tikz/matlab2tikz/wiki/FAQ#time-series-plot-raises-dimension-too-large-tex-error) is helpful.

**Comment by Pratched** [comment]
I will have a look tomorrow. Done for today ;)

**Comment by Pratched** [comment]
Well, I did had a look a few days ago, but havnt found time to implement this. I think I understood what youre doing and how its working.

Sadly I wasnt able to close this case...
Here is a picture how it looks in Matlab(right) and what Latex is giving me(left). You maay notice that the ylabel for ax(2) is on the left side, the legend misses one graph and is looking funny after all, there is a suspicious "10^5"  in the bottom right corner and the whol tkiz is squeezed.
<img width="698" alt="aufg2_figure" src="https://cloud.githubusercontent.com/assets/19487828/15633555/f68eb0da-25af-11e6-8c12-cf9d998c05ec.PNG">

I am using this script, it should work with the .mat-file I provided earlier.

```
load('Aufg2_Messdaten');

G_inW=G_inmV*10;
A_inm2=12*2.6*1.1;
Q_SinW=G_inW*A_inm2;
P_inW_DC=I_PVinmA.*U_PVinV/1000;
n_DC_real=P_inW_DC./Q_SinW;

[ax, ~ ,~] = plotyy(ZeitA, [G_inW, P_inW_DC], ZeitA, n_DC_real);
axis([7 8 0 2000]);
set(ax(2), 'YColor', 'black')
set(ax(1), 'YTick', [0 500 1000 1500 2000]); 
datetick(ax(1)); datetick(ax(2));
grid on;
legend(gca, 'string',{'$\dot G$','$P_{DC}$', '$\eta_{DC,real}$'},'interpreter','latex');
title('Tagesverlauf am 27.04.2016')
xlabel('Uhrzeit [hh:mm]'); 
ylabel(ax(1),'Leistung in [W]');
ylabel(ax(2), 'Wirkungsgrad');
```

I am sorry to bother you again, but maybe youve got some time and helpful ideas (once again). In the meantime I try to use the FAQ provided by okomarov, but I dont think, that this will be the problem here.

**Comment by miscco** [comment]
Hm, that is really strange, as it was working for me.

So for the axis you can add the following to set(ax(2))
```matlab
'YAxisLocation', 'right'
```

I dont know, why it does not account for the legend enty. @okomarov is the expert there. Also please make sure, that you are using the latest development version.

**Comment by Pratched** [comment]
Well, that was easy. Could have found out by myself, thanks a lot.

Did you got right today or last week? Cause I created a complete new tex-file only plotting the given file and it looks just the same. 
I have got a few other matlab-plots in the actual .tex-file and they are all working quite well

**Comment by Pratched** [comment]
Just fyi matlab2tikz is saying I do need pgfplots 1.4 or newer and Latex is suggesting to write `\pgfplotsset{compat=1.13}` which probably means compability?

**Comment by miscco** [comment]
Can you check your pfgplots version?

It is explained in the FAQ found here https://github.com/matlab2tikz/matlab2tikz/wiki/FAQ

Maybe you just need a newer version.

**Comment by Pratched** [comment]
Alrigth, I guess it is version 1.13 
<img width="709" alt="untitled" src="https://cloud.githubusercontent.com/assets/19487828/15633856/5fdcdd08-25b7-11e6-87a0-627635945203.png">

I removed Miktex and reinstalled the latest version but the output for checking the version pf pgfplots is the same. I dont know why miktex isnt installing the latest version of pgfplots.

**Comment by miscco** [comment]
1.13 is the most recent version, so that is fine ;)

**Comment by Pratched** [comment]
Alright, one thing that works ;)
The solution for ylabel you provided only works if i set compat=1.13. If I dont, both ylabels stay on the left side. Dont know if texmaker is using another version without the setting...

**Comment by Pratched** [comment]
The FAQ @okomarov suggested was quite helpfull. I needed to insert an extra xlim and ylim (for ax(1) and ax(2)) to get the xaxis synchronized. In my opinion it seems like the xvalues are bold, that might be okay. The only "important" problem is the legend. there seems to be some trouble left.

Ps: If i could get  xmin starting at 0300 and xmax ending with 2100 (like befor) that would be great, but I am asuming this is made up by 'unique'
<img width="497" alt="aufg2_figure" src="https://cloud.githubusercontent.com/assets/19487828/15634197/52ee24b0-25be-11e6-8948-a5af60dab088.PNG">
 
```

load('Aufg2_Messdaten');

G_inW=G_inmV*10;
A_inm2=12*2.6*1.1;
Q_SinW=G_inW*A_inm2;
P_inW_DC=I_PVinmA.*U_PVinV/1000;
n_DC_real=P_inW_DC./Q_SinW;


% Re-map dates and re-plot
%[~,~,x] = unique(ZeitA);

[ax, ~ ,~] = plotyy(x, [G_inW, P_inW_DC], x, n_DC_real);
    %legt für beide achsen jeweils eine Dateinummer(Handle) fest
axis(ax(1),[0 177 0 2000]);
axis(ax(2),[0 177 0 0.2]);

% Choose number of ticks and format of labels
numticks = 7;
fmt      = 'hh:MM';
% Set manually
ticks  = linspace(x(1),x (end),numticks);
labels = datestr(linspace(min(ZeitA),max(ZeitA),numticks), fmt);
set(ax(1), 'xtick',ticks, 'XTickLabel',labels)
set(ax(2), 'xtick',ticks, 'XTickLabel',labels)

set(ax(2), 'YColor', 'black','YAxisLocation', 'right')
set(ax(1), 'YTick', [0 500 1000 1500 2000]); 

%datetick(ax(1)); datetick(ax(2));
grid on;
legend(gca, 'string',{'$\dot G$','$P_{DC}$', '$\eta_{DC,real}$'},'interpreter','latex');
title('Tagesverlauf am 27.04.2016')
xlabel('Uhrzeit [hh:mm]'); 
ylabel(ax(1),'Leistung in [W]');
ylabel(ax(2), 'Wirkungsgrad');
```

**Comment by Pratched** [comment]
Well I do hope this is the last comment I have to write. I could fix the legend-problem by typing 
```
legend(ax(1), 'string',{'$\dot G$','$P_{DC}$'},'interpreter','latex');
legend(ax(2), 'string', { '$\eta_{DC,real}$'},'interpreter','latex');
```
Probably the same problem had with the synchronized values for the xaxis.

I do thank you all for helping me out!
If somebody has an idea for the xaxis not "being bold" and  the xmin(starting0300) and xmax(ending 2100) that would be great (Maybe I i need to add some values because I used `unique`). I would try any solution in a few days, but I need to get this task done at first.

**Comment by egeerardyn** [comment]
> Alright, one thing that works ;)
The solution for ylabel you provided only works if i set `compat=1.13`. If I dont, both ylabels stay on the left side. Dont know if texmaker is using another version without the setting...

The `compat=1.13` is something that needs to be set explicitly (and should be equivalent to `compat=newest` (until 1.14 comes out)). If you don't set the mode, I think [pgfplots](http://ctan.org/pkg/pgfplots) normally uses `1.3` for backwards compatibility reasons. This has nothing to do with the editor you use for LaTeX.

**Comment by miscco** [comment]
So your "Zeit" does not contain 3:00 and 21:00, so you would have to add those x-values and the corresponding number of NaNs to the yData.

Getting the correct time would be possible through
```matlab
ZeitA = [datenum('18:00'); ZeitA; datenum('21:00')];
```
and the three yData would get
```matlab
G_inW = [NaN; G_inW; NaN];
```

**Comment by Pratched** [comment]
Thanks a lot. 

PS: I am really sorry, but since you are very helpfull all the time maybe you could just tell me one more thing: Is it possible while running `matlab2tikz` to save the .tikz-file somewhere else than in the folder where matlab2tikz.m is stored? Maybe add something like matlab2tikz('mytikzfile.tikz', 'C:/wanted/folder/').

Well I did tried this one and a few other things, but I wasnt able to just "add to path" the matlab2tikz.m-file to the current folder (which would be C:/wanted/folder). 
I did manage with `cd` to get from C:/wanted/folder to C:/.../matlab2tikz and running matlab2tikz but this keeps saving the .tikz-file in C:/.../matlab2tikz.
I dont want to bother you anymore but I am looking about one hour by now. Maybe there is no way to do so and you can just keeping me from searching an longer.
Of course I could copy the matlab2tikz-folder into any C:/wanted/folder-directory but where would be fun in that ;)

**Comment by miscco** [comment]
@Pratched have a look at addpath() Then you can run m2t from everywhere. >>that works for any other script too.

Either you use the full path or do addpath(pdw) in the matlab2tikz folder.

**Comment by Pratched** [comment]
Okay, this is wired. I already add this folder to the search path permanently. Probably never tried to run the matlab2tikz.m-file in my script without any further `cd`...

Thank you again. You guys are doing a great job with matlab2tikz!

**Comment by miscco** [comment]
@Pratched make sure, that you have the full path not just relative ones

**Comment by Pratched** [comment]
I did;)
I acutally added the folder which is containing the m2t-folder earlier. But probably only this one without any subfolder.

**Comment by egeerardyn** [comment]
@Pratched:

> PS: I am really sorry, but since you are very helpfull all the time maybe you could just tell me one more thing: Is it possible while running matlab2tikz to save the .tikz-file somewhere else than in the folder where matlab2tikz.m is stored? Maybe add something like matlab2tikz('mytikzfile.tikz', 'C:/wanted/folder/').

As @miscco said,

```matlab
addpath(PATH_TO_MATLAB2TIKZ)
% with PATH_TO_MATLAB2TIKZ the absolute path to the folder containing matlab2tikz.m, e.g. 'C:\Users\Me\Documents\MATLAB\matlab2tikz\src'

% if it is already on your path, you should get the absolute path back from
PATH_TO_MATLAB2TIKZ = which('matlab2tikz')
``` 
should allow you to call `matlab2tikz` from where-ever you like. 
I recommend you either use [Set Path](http://nl.mathworks.com/help/matlab/matlab_env/add-remove-or-reorder-folders-on-the-search-path.html) or add the relevant `addpath` statement to your [startup script](http://nl.mathworks.com/help/matlab/ref/startup.html) (this is more advanced).

Saving the output file in any folder is as easy as:
```matlab
matlab2tikz( 'C:\Users\Me\Desktop\file.tikz', ...)
```

allows you to save the output to whichever directory you please.

And since you found those `cd ..` things, `matlab2tikz` can cope with paths such as `..\file.tikz` as input as well if you want to place the file in the underlying folder. This is in contrast to `addpath` where it is better NOT to use `..` anywhere in its argument (it can handle those, but you can produce strange effects in doing so).

**Comment by Pratched** [comment]
@okomarov @miscco guys I am sorry to bother you but I am not sure if you got notified and just have no idea how to fix this or I didnt reopen #910 . If so maybe you could have  a look, becaus I tried to set ticks for the FAQ (2 graphs in `plot`) @okomarov suggested but i couldnt  get it working.

I think I did not reopen the issue. I tried but there seems not to be an option for my profil.

**Comment by miscco** [comment]
Hm, i dont think that this is the right place to discuss general matlab problems.

Anyway, it seems that there is there is an error in your code, it should be
```matlab
% Set manually
ticks   = linspace(dt(1),dt(end),numticks);
labels = datestr(linspace(min(dt),max(dt),numticks), fmt);
set(gca, 'Xtick',ticks, 'XTickLabel',labels)
```

**Comment by Pratched** [comment]
yep sorry about that, but the probability to get some help for this is much higher here than in any other matlab forum I guess, since this is something i could easly get done with `datetick`. 
And maybe @okomarov wanted some feedback for the FAQ he provided.

But thank you for taking a look. Sadly it does not change anything. There are still no values for the x axis at all... But for this figure I will use  higher dpi and insert an png. This will be alright. for that matter.

**Comment by miscco** [comment]
@Pratched i understand that, but generally i would suggest against that. You are currently lucky that im in parental leave otherwise i wouldnt have that much time.

Seems like i copied the wrong code. The problem is, that you pass the position in the absolute time, yet plot the indices. Therefore the ticks are outside of your axis. The solution is to put them in the space of the number of data points.

```matlab
ticks   = linspace(1,numel(dt),numticks);
labels = datestr(linspace(min(dt),max(dt),numticks), fmt);
set(gca, 'Xtick',ticks, 'XTickLabel',labels)
```

However, this has the disadvantage, that you might get uneven ticks. In that cae you have to get the indices of the original ticks and use them.

**Comment by Pratched** [comment]
And I really appreciate this! You helped me a lot. In the future I will use common matlab forums (and start earlier). 
And I am grateful for providing matlab2tikz and teking care of it.

b2t: this fixes my problem. uneven ticks arent a problem for me here. So this is fine. Thank you!

**Comment by miscco** [comment]
@Pratched You are welcome. If you have any other question regarding m2t dont hesitate to come here

**Comment by Pratched** [comment]
I will ;)
